python/lsst/pipe/drivers/background.py

```diff
@@ -0,0 +1,718 @@
+        self._values = values
+        if numbers is None:
+            numbers = afwImage.ImageF(self.dims)  # float for dynamic range and convenience
+            numbers.set(0.0)
+        else:
+            numbers = numbers.clone()
+        assert(numbers.getDimensions() == self.dims)
+        self._numbers = numbers
+
+    def __reduce__(self):
+        return self.__class__, (self.config, self.dims, self.transform, self._values, self._numbers)
+
+    def clone(self):
+        return self.__class__(self.config, self.dims, self.transform, self._values, self._numbers)
+
+    def addCcd(self, exposure):
+        """Add CCD to model
+
+        We measure the background on the CCD (clipped mean), and record
+        the results in the model.  For simplicity, measurements are made
+        in a box on the CCD corresponding to the warped coordinates of the
+        superpixel rather than accounting for little rotations, etc.
+        We also record the number of pixels used in the measurement so we
+        can have a measure of confidence in each bin's value.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            CCD exposure to measure
+        """
+        detector = exposure.getDetector()
+        transform = detector.getTransformMap().getTransform(detector.makeCameraSys(afwCameraGeom.PIXELS),
+                                                            detector.makeCameraSys(afwCameraGeom.FOCAL_PLANE))
+        image = exposure.getMaskedImage()
+        maskVal = image.getMask().getPlaneBitMask(self.config.mask)
+
+        # Warp the binned image to the focal plane
+        toSample = transform.then(self.transform)  # CCD pixels --> focal plane --> sample
+
+        warped = afwImage.ImageF(self._values.getBBox())
+        warpedCounts = afwImage.ImageF(self._numbers.getBBox())
+        width, height = warped.getDimensions()
+
+        stats = afwMath.StatisticsControl()
+        stats.setAndMask(maskVal)
+        stats.setNanSafe(True)
+        # Iterating over individual pixels in python is usually bad because it's slow, but there aren't many.
+        pixels = itertools.product(range(width), range(height))
+        for xx, yy in pixels:
+            llc = toSample.applyInverse(afwGeom.Point2D(xx - 0.5, yy - 0.5))
+            urc = toSample.applyInverse(afwGeom.Point2D(xx + 0.5, yy + 0.5))
+            bbox = afwGeom.Box2I(afwGeom.Point2I(llc), afwGeom.Point2I(urc))
+            bbox.clip(image.getBBox())
+            if bbox.isEmpty():
+                continue
+            subImage = image.Factory(image, bbox)
+            result = afwMath.makeStatistics(subImage, afwMath.MEANCLIP | afwMath.NPOINT, stats)
+            mean = result.getValue(afwMath.MEANCLIP)
+            num = result.getValue(afwMath.NPOINT)
+            if not numpy.isfinite(mean) or not numpy.isfinite(num):
+                continue
+            warped.set(xx, yy, mean*num)
```

Comment:
I was referring to this in the previous comment. Let's say `num << npixels` then you get a significantly smaller value here. I thought that instead you wanted to pretend that all pixels in the box had the mean value. But apparently that is not the case?

Reply:
I'm pretending that `num` pixels have the same value.
Pixels from a different CCD may overlap this superpixel, so those are going to get included as well, weighted by the number of pixels.